src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+        "This is also useful for adding trusted artifact tasks, e.g. name the task as "
+        "sast-coverity-check but the actual task name is sast-coverity-check-oci-ta.",
+    )
+    add_task_parser.add_argument(
+        "-a",
+        "--run-after",
+        metavar="TASK_NAME",
+        dest="run_after",
+        action="append",
+        help="Name of task running before the adding task. "
+        "This is the task name used in the build pipeline definition. "
+        "This argument can be specified multiple times to add more than one tasks.",
+    )
+    add_task_parser.add_argument(
+        "-p",
+        "--param",
+        type=task_param,
+        metavar="PARAM",
+        dest="params",
+        action="append",
+        help="Specify a task parameter that consists of comma-separated name and value. "
+        "This argument can be specified multiple times to add more than one parameters.",
+    )
+    add_task_parser.add_argument(
+        "-s",
+        "--skip-checks",
+        action="store_true",
+        dest="skip_checks",
+        help="Skip this task if it can be skipped as a check for a fast build.",
+    )
+    add_task_parser.add_argument(
+        "-r",
+        "--bundle-ref",
+        type=konflux_task_bundle_reference,
+        metavar="IMAGE_REF",
+        dest="bundle_ref",
+        help="Use Tekton bundle resolver to reference the expected task bundle. "
+        "The full reference has to include both tag and digest, "
+        "e.g. registry/org/task-name:tag@digest."
+        "If omitted, the latest task bundle is queried from the registry.",
+    )
+    add_task_parser.add_argument(
+        "-g",
+        "--git-add",
+        dest="git_add",
+        action="store_true",
+        help="Add the modified files to git index.",
+    )
+    add_task_parser.set_defaults(action=action)
+
+
+def action(args) -> None:
+    actual_task_name: Final = args.task
+    pipeline_task_name: Final = args.pipeline_task_name or args.task
+
+    bundle_ref = args.bundle_ref
+    if not bundle_ref:
+        bundle_ref = KonfluxBuildDefinitions().query_latest_bundle(actual_task_name)
+
+    logger.info("Adding task %s, bundle %s", actual_task_name, bundle_ref)
+
+    task_config = {
+        "name": pipeline_task_name,
+        "taskRef": {
+            "resolver": "bundles",
+            "params": [
+                {"name": "kind", "value": "task"},
+                {"name": "name", "value": actual_task_name},
+                {"name": "bundle", "value": bundle_ref},
+            ],
+        },
+    }
+    if args.params:
+        task_config["params"] = [{"name": name, "value": value} for name, value in args.params]
+    if args.run_after:
+        task_config["runAfter"] = args.run_after
+    if args.skip_checks:
+        task_config["when"] = [
+            {"input": "$(params.skip-checks)", "operator": "in", "values": [False]}
```

Comment:
The `False` should be a string `"false"` I think, Tekton doesn't support boolean params